### ISSUE TITLE
reportMissingReturnType not catching functions without return type annotations

### ISSUE THREAD

**Issue by zhupanov** [microsoft/pyright#9772]
The `reportMissingReturnType` setting appears not to be working for functions that completely lack return type annotations, while it correctly catches incorrect return type annotations.

Minimal reproduction:

1. Config (pyrightconfig.json):
```json
{
  "reportMissingReturnType": "error",
  "analyzeUnannotatedFunctions": true
}
```
Test code:
```Python
def test1():    # Should error: no return type annotation
    return 42

def test2() -> None:    # Does error as expected: returns int but annotated None
    return 42

def test3() -> int:    # Correct, no error expected or seen
    return 42
```
Expected behavior:

Error for test1() (missing return type)
Error for test2() (incorrect return type)
No error for test3()

Actual behavior:

No error for test1() ❌
Error for test2() ✅
No error for test3() ✅

Environment:

pyright version: 1.1.391
Python version: 3.12
Platform: macOS

This might be related to previously fixed issue #2689 and currently open issue #6290.

**Comment by erictraut** [comment]
Pyright does not have a `reportMissingReturnType` diagnostic check. This is intentional. Missing return types do not affect type safety because pyright is able to infer return types from `return` statements in the function. See [this issue](https://github.com/microsoft/pyright/issues/3013) for full details.

See [this documentation](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-settings-defaults) for a full list of supported diagnostic checks. 

Pyright offers a `--verifytypes` option that you can run on a library to ensure that all functions and methods that comprise its public interface have explicit return types.

I'm guessing that you're using an unauthorized fork of pyright.
